packages/@lwc/ssr-runtime/src/index.ts

```diff
@@ -210,74 +181,116 @@ export class LightningElement implements PropsAvailableAtConstruction {
         this.__attrs[attrName] = value;
     }
 
+    hasAttribute(attrName: string): boolean {
+        return Boolean(this.__attrs && attrName in this.__attrs);
```

Comment:
```suggestion
        return Boolean(this.__attrs && attrName in this.__attrs && this.__attrs[attrName] !== null);
```

Technically you can `setAttribute('foo', null)` and it's the same as removing it.

Reply:
Opted to go the other way -- `setAttribute('foo', null)` will now remove the attribute from `__attrs`.

Reply:
Whoops I was wrong on this! `getAttribute` returns null but `setAttribute` sets to string `"null"`. 😆